### ISSUE TITLE
Fix `selector-type-no-unknown` false positives for idents in functional pseudo-classes

### ISSUE THREAD

**Issue by elskhn** [stylelint/stylelint#8189]
### What minimal example or steps are needed to reproduce the bug?

```css
html:active-view-transition-type(left) {
  animation: none;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-element-no-unknown": true
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIAWAXAtgG3gQzCglgG4wC0h+MA7iSgE7YB2AzvgRAzQJ4AOMAFBhgAzFAEoABMAA6DceMb402Ng3jiG7GAG4ZAXxAAaEEPyCActjRxEMAB6XuggHRgmTQ+HYmA5ghDTZcSkQOxQYBigmYLUAbRk5OWCmFE5BDHwGFBJIBh8SZMYobFooYPjxAF09DxyfADEIWiUUPwArJnYPWG53RADEkGTUmHTM6KCQDGUYZOCDcqSUtIys2vxvfJRC4tKECamw2ZBq3SA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "16.11.0",
  "stylelint-use-nesting": "6.0.0",
}
```

### What did you expect to happen?

No problems to be reported.

### What actually happened?

The following problem was reported: 
```
  708:38  ✖  Unexpected unknown type selector "slide-left"   selector-type-no-unknown
```

### Do you have a proposal to fix the bug?

Pleas add support for newer View Transition API selectors such as `active-view-transition-type(*)`.

Chrome article explaining this selector's usage: https://developer.chrome.com/docs/web-platform/view-transitions/same-document#view-transition-types

**Comment by jeddy3** [comment]
@elskhn Thanks for the report and for using the template.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

It should be just a case of replicating what we already do for similar pseudo-elements, but for pseudo-classes:

https://github.com/stylelint/stylelint/blob/74f3c080f20302c724659aaac24d47fcc713ab49/lib/rules/selector-type-no-unknown/index.mjs#L29-L35

[Spec for reference](https://drafts.csswg.org/css-view-transitions-2/#the-active-view-transition-type-pseudo) for those curious:

> The `:active-view-transition-type()` pseudo-class applies to the root element of the document, if it has a matching [active view transition](https://drafts.csswg.org/css-view-transitions-1/#document-active-view-transition). It has the following syntax definition: `:active-view-transition-type`([`<custom-ident>`](https://drafts.csswg.org/css-values-4/#identifier-value)[#](https://drafts.csswg.org/css-values-4/#mult-comma))

**Comment by Mouvedia** [comment]
The new set will be:

```js
 const IGNORED_PSEUDO_CLASSES = new Set([ 
  ':active-view-transition-type',
  ':dir',
  ':-moz-locale-dir',
  ':state',
  ':lang',
 ]); 
```

ref #7915

**Comment by elskhn** [comment]
Perfect, I'll open a PR with that change

**Comment by Mouvedia** [comment]
off-topic

@jeddy3 should we add `::part()` is a separate issue?

**Comment by elskhn** [comment]
It seems [this PR](https://github.com/stylelint/stylelint/pull/7868/files) was supposed to fix this issue. Any idea why it doesn't work?

**Comment by Mouvedia** [comment]
@elskhn as you can see in its title, this PR was for `selector-pseudo-class-no-unknown` and yours is for `selector-type-no-unknown`.

**Comment by elskhn** [comment]
^ ah, of course

I've opened [a PR](https://github.com/stylelint/stylelint/pull/8191) to resolve this issue – please take a look when you have a chance
